tests/test_events_api.py

```diff
@@ -0,0 +1,172 @@
+from os import environ
+from typing import Callable
+from unittest.mock import MagicMock, patch
+
+from lbz.aws_boto3 import Boto3Client
+from lbz.events.api import BaseEvent, EventAPI
+
+
+class MyTestEvent(BaseEvent):
+    type = "MY_TEST_EVENT"
+
+
+class TestBaseEvent:
+    def test_base_event_creation_and_structure(self) -> None:
+        new_event = MyTestEvent({"x": 1})
+
+        assert hasattr(new_event, "raw_data")
+        assert hasattr(new_event, "data")
+        assert new_event.type == "MY_TEST_EVENT"
+        assert new_event.raw_data == {"x": 1}
+        assert new_event.data == '{"x": 1}'
+
+
+class TestEventApi:
+    def setup_method(self) -> None:
+        # pylint: disable= attribute-defined-outside-init
+        with patch.dict(environ, {"AWS_LAMBDA_FUNCTION_NAME": "million-dollar-lambda"}):
+            self.event_api = EventAPI()
+
+    def teardown_method(self, _test_method: Callable) -> None:
+        # pylint: disable= attribute-defined-outside-init
+        self.event_api._del()  # type: ignore # pylint: disable=protected-access
+
+    def test_init_with_defaults(self) -> None:
+        # pylint: disable=protected-access
+        assert self.event_api._pending_events == []
+        assert self.event_api._sent_events == []
+        assert self.event_api._source == "million-dollar-lambda"
+        assert self.event_api._bus_name == "million-dollar-lambda-event-bus"
+        assert self.event_api._resources == []
+
+    def test___repr__(self) -> None:
+        expected_repr = (
+            "<EventAPI bus: million-dollar-lambda-event-bus Events: pending=0 sent=0 failed=0>"
+        )
+        assert str(self.event_api) == expected_repr
+        self.event_api.register(MyTestEvent({"x": 1}))
+        self.event_api.register(MyTestEvent({"x": 1}))
+        expected_repr = (
+            "<EventAPI bus: million-dollar-lambda-event-bus Events: pending=2 sent=0 failed=0>"
+        )
+        assert str(self.event_api) == expected_repr
+        self.event_api._mark_sent()  # pylint: disable=protected-access
+        expected_repr = (
+            "<EventAPI bus: million-dollar-lambda-event-bus Events: pending=0 sent=2 failed=0>"
+        )
+        assert str(self.event_api) == expected_repr
+        self.event_api.register(MyTestEvent({"x": 1}))
+        self.event_api.register(MyTestEvent({"x": 1}))
+        self.event_api._mark_failed()  # pylint: disable=protected-access
```

Comment:
Ok, testing the representation might be useful but this test should be as simple as possible, definitely not containing three separate scenarios that base on the protected methods...

Reply:
This class has two major public functions, `register()` and `send()` - test them deeply instead of testing everything around. Thanks to that you will probably achieve a shorter test module that will be more valuable in the future 😉